src/helpers.php

```diff
@@ -18,3 +18,22 @@ function getModelForGuard(string $guard)
             })->get($guard);
     }
 }
+
+if (! function_exists('currentAuthGuard')) {
+    /**
+     *
+     * @return string|null
+     */
+    function currentAuthGuard()
+    {
+        if(! config('permission.enable_dynamic_auth_guard_checks')) {
+            return null;
+        }
+
+        foreach (array_keys(config('auth.guards')) as $guard) {
+
+            if (auth()->guard($guard)->check()) return $guard;
+        }
+        return null;
+    }
```

Comment:
If I recall correctly the reason why the Guard model's lookup function checks not only for the array_key but also for a valid (defined) provider is because Lumen does things somewhat differently from Laravel here, and without it Lumen wasn't compatible.

Reply:
To be honest I copied that check from an other issue.
I didn't find an elegant way to get the active guard, is laravel missing a piece here?

Reply:
> I didn't find an elegant way to get the active guard, is laravel missing a piece here?

The only way is to read config(auth.guard.default): Laravel sets (replaces) it with the name of the active guard when the guard changes.